### ISSUE TITLE
Investigate supporting FlightSQL

### ISSUE THREAD

**Issue by matthewmturner** [datafusion-contrib/datafusion-tui#81]

**Comment by backkem** [comment]
Note that there are FlightSQL table provider implementations:
* [datafusion-contrib/datafusion-table-provider-flight](https://github.com/datafusion-contrib/datafusion-table-provider-flight): pure table provider.
* [datafusion-contrib/datafusion-federation flight-sql example](https://github.com/datafusion-contrib/datafusion-federation/blob/ce87bf45681d230e816f27e545e720498c2de955/examples/examples/flight-sql.rs): FlightSQL federation. Allowing whole parts of the logical plan to be sent to the remote as SQL or Substrait.

(Note: the above may be merged and moved to [datafusion-contrib/datafusion-table-providers](https://github.com/datafusion-contrib/datafusion-table-providers/), see [this comment](https://github.com/datafusion-contrib/datafusion-federation/issues/43#issuecomment-2304083274).

**Comment by matthewmturner** [comment]
@backkem Noted, thanks for pointing that out. 

My original goal with FlightSQL was to embed a client and have displays for metadata / catalog information.  I don't think that and what you mention are mutually exclusive though.

**Comment by backkem** [comment]
Indeed, these TableProvider implementations allow you to query via FlightSQL. I'm not sure if catalog inspection is implemented yet but that would also make sense.

**Comment by matthewmturner** [comment]
@backkem FYI there is a FlightSQL client now.  Next week I should be able to look into the flightsql `TableProviderFactory`
